haystack_experimental/components/agents/agent.py

```diff
@@ -2,29 +2,64 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
-import inspect
+# ruff: noqa: I001
+
+from dataclasses import dataclass
 from typing import Any, Optional, Union
 
+# Monkey patch Haystack's AgentSnapshot with our extended version
+import haystack.dataclasses.breakpoints as hdb
+
+from haystack_experimental.dataclasses.breakpoints import AgentSnapshot
+
+# replace reference
+hdb.AgentSnapshot = AgentSnapshot
```

Comment:
@anakin87 Could you comment on whether this is a good idea or not ...I remember you doing something similar for `ChatMessage`.

I am doing this here to avoid needing to bring even more functions from main to experimental but I have a feeling it may not be worth the linting and typing headache.